Add D1 changelog and release notes for request latency improvements (#19235)
cloudflare/cloudflare-docs: src/content/changelogs-next/2025-01-07-d1-faster-query.mdx

---
title: 40-60% Faster D1 Requests
description: D1 end-to-end request latency decreases by 40-60%.
products:
  - d1
date: 2025-01-07T18:00:00Z
---
 

> [D1](/d1/) lowered end-to-end request latency by 40-60% by eliminating latency incurred with multiple network round trips required for each request. Faster D1 performance applies to requests via both D1's [Worker API](/d1/worker-api/) and [REST API](/api/resources/d1/subresources/database/methods/query/).

Reviewer: ```suggestion
[D1](/d1/) lowered end-to-end request latency by 40-60% by eliminating redundant network round-trips for each request. Faster D1 performance applies to requests via both D1's [Worker API](/d1/worker-api/) and [REST API](/api/resources/d1/subresources/database/methods/query/), although the above speedup will be mostly observed when using the D1 [Worker API](/d1/worker-api/).
```

Reviewer 2: Agree, though I'd change:
`Faster D1 performance applies mostly to requests via D1's [Worker API](/d1/worker-api/).`
to
`Faster D1 performance is most noticeable via requests to D1's [Worker API](/d1/worker-api/).`

> Removing netowrk round trips particularily benefit user requests geographically far from the requested D1 database, e.g, user in Europe talking to a database in North America. D1 [location hints](https://developers.cloudflare.com/d1/configuration/data-location/#provide-a-location-hint) can be used to influence the geographic location of a database.

Reviewer: This speedup applies to everyone using D1, especially those without smart placement. I wouldn't want to mislead folks thinking that unless they are cross-region they won't get any benefit.

Author: Will clarify that all benefit but that the highest impact would be cross region

Reviewer: @vy-ton I don't agree that it's necessarily cross-region mostly. The impact is the same cross-region or not, , especially since Cloudflare's locations within a single region can be 50-100ms apart, so 2x round-trips could be 100ms+, which is substantial. I would keep it generic for the Workers API at 40-60% improvement.

Author: `keep it generic for the Workers API at 40-60% improvement.` - we are saying this. This docs page of mini blog updates should provide more user orientation/context.

If you have cross-region, then its 2x round trips where each round trip is 100ms+.

Author: LMK your thoughts based on the recent change.

> [TODO image]

Author: @joshthoward @lambrospetrou @rozenmd Is there a sanitized metrics graph we could share? We need one to show not tell a performance improvement.

FYI `changelog-next` is a recent page where we can share more ad-hoc product updates. Product specific changelog will likely get rebranded as release notes.

Reviewer: Yes, will share internally.

---
title: 40-60% Faster D1 Worker API Requests
description: D1's Worker API request latency decreases by 40-60%.
products:
  - d1
date: 2025-01-07T18:00:00Z
---
 

> [D1](/d1/) lowered end-to-end Worker API request latency by 40-60% by eliminating redundant network round trips for each request.

Reviewer 3: ```suggestion
[D1](/d1/) lowered end-to-end request latency via the [Workers API](/d1/worker-api/) by up to 60% through eliminating redundant network round trips for each request.
```

> ![D1 Worker API latency](~/assets/images/d1/faster-d1-worker-api.png)

Reviewer 3: Does this path work?

Author: @Oxyjun need your help with the error preventing the preview to confirm path for image assets

> ![D1 Worker API latency](~/assets/images/d1/faster-d1-worker-api.png)

Reviewer: ```suggestion
![D1 Worker API latency](../../assets/images/d1/faster-d1-worker-api.png)
```
Try this. Relative links should work and they auto-bundled by Starlight/Astro. https://starlight.astro.build/guides/authoring-content/#images

*p50, p90, and p95 latency aggregated across entire D1 service. These latencies are a reference point and should not be viewed as your exact workload improvement.

This performance improvement benefits all D1 Worker API traffic, especially cross-region requests where network latency is an outsized latency factor. For example, a user in Europe talking to a database in North America. D1 [location hints](/d1/configuration/data-location/#provide-a-location-hint) can be used to influence the geographic location of a database.

> For more details on how D1 lowered latency, see product-specific changelog [entry](/d1/platform/changelog/#2025-01-07).

Reviewer 3: ```suggestion
For more details on how D1 lowered latency, see the D1 specific [changelog entry](/d1/platform/changelog/#2025-01-07).
```

Author: accepted

---
title: 40-60% Faster D1 Worker API Requests
description: D1's Worker API request latency decreases by 40-60%.
products:
  - d1
date: 2025-01-07T18:00:00Z
---
 
[D1](/d1/) lowered end-to-end request latency via the [Workers API](/d1/worker-api/) by up to 60% through eliminating redundant network round trips for each request.

> ![D1 Worker API latency](../../assets/images/d1/faster-d1-worker-api.png)

Reviewer: Hmm, I don't see the image in https://bb52b71a.cloudflare-docs-7ou.pages.dev/changelog/

Reviewer: BTW, why not put the image in both entries? Makes the impact more obvious.

> ![D1 Worker API latency](../../assets/images/d1/faster-d1-worker-api.png)

Reviewer 4: ```suggestion
![D1 Worker API latency](~/assets/images/d1/faster-d1-worker-api.png)
```

Reviewer: @Oxyjun Is th `~/` aliased somewhere to refer to the `/public/images` folder?

*p50, p90, and p95 latency aggregated across entire D1 service. These latencies are a reference point and should not be viewed as your exact workload improvement.*

This performance improvement benefits all D1 Worker API traffic, especially cross-region requests where network latency is an outsized latency factor. For example, a user in Europe talking to a database in North America. D1 [location hints](/d1/configuration/data-location/#provide-a-location-hint) can be used to influence the geographic location of a database.

> For more details on how D1 removed redundnat round trips, see the D1 specific changelog [entry](/d1/platform/changelog/#2025-01-07).

Author: @irvinebroque I disagree. I want discoverability and navigation between `changelog-next` (product updates) and `D1 changelog` (release notes). A user skimming `changelog-next` entries should take away what changed and why does it affect me, without necessarily caring about the how.

In an ideal future:
- `product changelog` would be generated release notes from code rollouts
- `changelog-next` would be product updates from code rollouts/shipped features that are worth calling out

> For more details on how D1 removed redundnat round trips, see the D1 specific changelog [entry](/d1/platform/changelog/#2025-01-07).

Reviewer 5: your call! just saying that i initially interpreted the changelog post as, "whoa the _query_ times are slow!" and then only after understanding "we got rid of network roundtrips" did it click that — "ohhh the total time it takes to get data back from D1 over the network is what is getting faster, we're not talking about timing of D1 queries themselves"

> For more details on how D1 removed redundnat round trips, see the D1 specific changelog [entry](/d1/platform/changelog/#2025-01-07).

Author: good feedback, let me try to clarify/make the improvement more clear without full details

  - d1
date: 2025-01-07T18:00:00Z
---

> Users making [D1](/d1/) requests via the [Workers API](/d1/worker-api/) can see up to a 60% end-to-end latency improvement due to the removal of network round trips needed for each request to a D1 database.

Reviewer: `removal of EXTRA/redundant network round trips`
